cpp/src/arrow/compute/kernels/scalar_arithmetic_test.cc

```diff
@@ -149,7 +173,23 @@ template <typename T>
 class TestUnaryArithmeticUnsigned : public TestUnaryArithmeticIntegral<T> {};
 
 template <typename T>
-class TestUnaryArithmeticFloating : public TestUnaryArithmetic<T> {};
+class TestUnaryArithmeticFloating : public TestUnaryArithmetic<T, ArithmeticOptions> {};
+
+template <typename T>
+class TestUnaryArithmetic<T, SignOptions>
+    : public TestBaseUnaryArithmetic<T, SignOptions> {};
+
+template <typename T>
+class TestUnaryArithmeticIntegral_Sign : public TestUnaryArithmetic<T, SignOptions> {};
```

Comment:
Can you use `CamelCase` without exceptions?

Reply:
This could be called e.g. `TestSignIntegral`, and the classes below `TestSignUnsigned`, etc.

Reply:
Oops! Thanks for catching this.